server/controllers/admin/functions.js

```diff
@@ -0,0 +1,147 @@
+/**
+* Function Controller
+*
+* This controller exposes an API to the client for reading and writing Function
+
+*/
+var db = require('../../lib/db');
+
+// GET /Function
+function lookupFunction(id, codes) {
+  'use strict';
+
+  var record;
+
+  var sql =
+    'SELECT id, fonction_txt FROM fonction ' +
+    'WHERE fonction.id = ? ';
+
+  return db.exec(sql, [id])
+  .then(function (rows) {
+
+    if (rows.length === 0) {
+      throw new codes.ERR_NOT_FOUND();
+    }
+
+    // store the record for return
+    record = rows[0];
```

Comment:
Why not just `return rows[0];` and then you never have to initialize a new variable?